tests/sweep_framework/sweeps/ccl/all_gather_t3k_non_sharded.py

```diff
@@ -0,0 +1,216 @@
+
+    if tensor_layout_key == "tile":
+        input_shape = tile_input_shapes
+    else:
+        input_shape = row_input_shapes
+
+    param_name = f"all_gather_non_sharded_{tensor_layout_key}_{buffer_key}_{gather_op_str}_{dtype_str}_{orientation_str}_dim{dim}_{num_device}devices_{num_link}links"
+    return {
+        "name": param_name,
+        "config": {
+            "num_devices": [num_device],
+            "num_links": [num_link],
+            "input_shape": input_shape,
+            "dim": [dim],
+            "tensor_layout": [tensor_layout_value],
+            "input_dtype": [dtype],
+            "mem_config": [buffer_value],
+            "all_gather_topology": [gather_op],
+            "num_iters": [1],
+        },
+    }
+
+
+parameters = {}
+
+for dtype_str in dtype_map.keys():
+    for orientation_str in orientation_map.keys():
+        for gather_op_str in gather_op_map.keys():
+            for tensor_layout_key in tensor_layouts:
+                for buffer_key in buffer_types:
+                    for dim in dims:
+                        for num_device in num_devices:
+                            for num_link in num_links:
+                                param = generate_params(
+                                    dtype_str,
+                                    orientation_str,
+                                    gather_op_str,
+                                    tensor_layout_key,
+                                    buffer_key,
+                                    dim,
+                                    num_device,
+                                    num_link,
+                                )
+                                parameters[param["name"]] = param["config"]
+
+
+def invalidate_vector(test_vector) -> Tuple[bool, Optional[str]]:
+    unchunked_input_shape = list(test_vector["input_shape"])
+    unchunked_input_shape[test_vector["dim"]] *= test_vector["num_devices"]
+    if test_vector["num_devices"] < 2:
+        return True, f"Requires multiple devices to run"
+
+    if unchunked_input_shape[test_vector["dim"]] % test_vector["num_devices"] != 0 or (
+        test_vector["dim"] == 3 and unchunked_input_shape[test_vector["dim"]] // test_vector["num_devices"] % 32 != 0
+    ):
+        return True, f"Unsupported test case"
+    if test_vector["tensor_layout"] == ttnn.ROW_MAJOR_LAYOUT and test_vector["input_dtype"] == ttnn.bfloat8_b:
+        return True, f"bfloat8_b/4_b only supports TILE layout"
+    if test_vector["num_links"] == 2 and test_vector["num_devices"] == 8:
+        return True, f"8 devices and 2 links are not supported on t3000 devices"
+    return False, None
+
+
+def mesh_device_fixture():
+    assert ttnn.get_num_devices() >= 8, "Not T3000!"
+    device_ids = ttnn.get_t3k_physical_device_ids_ring()
+    num_devices_requested = len(device_ids)
+    mesh_device = ttnn.open_mesh_device(ttnn.MeshShape(1, num_devices_requested), mesh_type=ttnn.MeshType.Line)
```

Comment:
If this is intended to be a used for ring/line context, then I would just change this to:
```
mesh_device = ttnn.open_mesh_device(ttnn.MeshShape(2, 4), mesh_type=ttnn.MeshType.Ring)
```

because devices connected in a ring is also connected in a line.